pkg/api/handlers/mcpwebhookvalidation.go

```diff
@@ -0,0 +1,180 @@
+		Type:     gptscript.CredentialTypeTool,
+		Env:      secretCred,
+	}); err != nil {
+		_ = req.Delete(&validation)
+		return fmt.Errorf("failed to create credential: %w", err)
+	}
+
+	return req.Write(convertMCPWebhookValidation(validation, secretCred))
+}
+
+func (m *MCPWebhookValidationHandler) Update(req api.Context) error {
+	var validation v1.MCPWebhookValidation
+	if err := req.Get(&validation, req.PathValue("mcp_webhook_validation_id")); err != nil {
+		return err
+	}
+
+	var manifest types.MCPWebhookValidationManifest
+	if err := req.Read(&manifest); err != nil {
+		return types.NewErrBadRequest("failed to read manifest: %v", err)
+	}
+
+	if err := manifest.Validate(); err != nil {
+		return types.NewErrBadRequest("invalid manifest: %v", err)
+	}
+
+	secretCred := make(map[string]string, len(manifest.Webhooks))
+	for i, webhook := range manifest.Webhooks {
+		secretCred[strconv.Itoa(i)] = webhook.Secret
+		// Don't save the secrets in the database.
+		webhook.Secret = ""
+		manifest.Webhooks[i] = webhook
+	}
+
+	validation.Spec.Manifest = manifest
+
+	if err := req.Update(&validation); err != nil {
+		return fmt.Errorf("failed to update mcp webhook validation: %w", err)
+	}
+
+	// The only way to update a credential is to delete it and recreate it.
+	if err := req.GPTClient.DeleteCredential(req.Context(), system.MCPWebhookValidationCredentialContext, validation.Name); err != nil && !errors.As(err, &gptscript.ErrNotFound{}) {
+		return fmt.Errorf("failed to delete credential: %w", err)
+	}
+
+	if err := req.GPTClient.CreateCredential(req.Context(), gptscript.Credential{
+		Context:  system.MCPWebhookValidationCredentialContext,
+		ToolName: validation.Name,
+		Type:     gptscript.CredentialTypeTool,
+		Env:      secretCred,
+	}); err != nil {
+		return fmt.Errorf("failed to create credential: %w", err)
+	}
+
+	return req.Write(convertMCPWebhookValidation(validation, secretCred))
+}
+
+func (m *MCPWebhookValidationHandler) Delete(req api.Context) error {
+	var validation v1.MCPWebhookValidation
+	if err := req.Get(&validation, req.PathValue("mcp_webhook_validation_id")); err != nil {
+		return err
+	}
+
+	secretCred, err := req.GPTClient.RevealCredential(req.Context(), []string{system.MCPWebhookValidationCredentialContext}, validation.Name)
+	if err != nil {
+		return fmt.Errorf("failed to reveal credential: %w", err)
+	}
+
+	if err = req.GPTClient.DeleteCredential(req.Context(), system.MCPWebhookValidationCredentialContext, validation.Name); err != nil && !errors.As(err, &gptscript.ErrNotFound{}) {
+		return fmt.Errorf("failed to delete credential: %w", err)
+	}
+
+	if err = req.Delete(&validation); err != nil {
+		return fmt.Errorf("failed to delete mcp webhook validation: %w", err)
+	}
+
+	return req.Write(convertMCPWebhookValidation(validation, secretCred.Env))
+}
+
+func convertMCPWebhookValidation(validation v1.MCPWebhookValidation, secrets map[string]string) types.MCPWebhookValidation {
+	for key, secret := range secrets {
```

Comment:
Do we want to actually expose the secret in api?

Reply:
I would avoid exposing that unless client really needs the secret

Reply:
I will be changing this after our standup discussion.